plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectPushdownWithSplits.java

```diff
@@ -0,0 +1,101 @@
+ */
+package io.trino.plugin.hive.s3select;
+
+import io.airlift.units.DataSize;
+import io.trino.plugin.hive.AbstractTestHiveFileSystemS3;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.spi.connector.SchemaTableName;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.CSV_SCAN_RANGE_FILES;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.CSV_SCAN_RANGE_TYPES;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.PIPE_SPLIT_DELIMITER;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+
+public class TestHiveFileSystemS3SelectPushdownWithSplits
+        extends AbstractTestHiveFileSystemS3
+{
+    private String host;
+    private int port;
+    private SchemaTableName tableWith3MBSplits;
+    private int maxSplitSizeMB = 1;
+    private int maxInitialSplitSizeMB = 1;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.database = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+        this.tableWith3MBSplits = new SchemaTableName(database, "trino_s3select_test_csv_scan_range_pushdown");
+
+        // Initial setup is needed for running the inherited tests
+        super.setup(host, port, database, awsAccessKey, awsSecretKey, writableBucket, testDirectory, true);
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{3, 2}, {50, 30}};
+    }
+
+    @Override
+    protected void initHiveConfig(boolean s3SelectPushdownEnabled)
+    {
+        config = new HiveConfig()
+                .setS3SelectPushdownEnabled(s3SelectPushdownEnabled)
+                .setMaxSplitSize(DataSize.of(maxSplitSizeMB, MEGABYTE))
+                .setMaxInitialSplitSize(DataSize.of(maxInitialSplitSizeMB, MEGABYTE));
+    }
+
+    @Test(dataProvider = "testSplitSize")
+    public void testQueryPushdownWithSplitSize(int maxSplitSizeMB, int maxInitialSplitSizeMB)
+            throws Exception
+    {
+        this.maxSplitSizeMB = maxSplitSizeMB;
```

Comment:
Same here. Also the flow somehow tricky to follow. This method modifies certain fields from a parent class for them to be picked up by a method from parent class invoked later on. I would consider refactoring it to simplify the flow.